src/ui/components/SignTransactionButton/SignTransactionButton.tsx

```diff
@@ -19,16 +18,17 @@ import {
 } from 'src/ui/ui-kit/Button';
 import CheckIcon from 'jsx:src/ui/assets/checkmark-checked.svg';
 import { HStack } from 'src/ui/ui-kit/HStack';
+import type { OneOf } from 'src/shared/type-utils/OneOf';
+import type { SubmittedTransactionResponse } from 'src/shared/types/SubmittedTransactionResponse';
+import type { SolTransaction } from 'src/modules/solana/SolTransaction';
+import { solToBase64 } from 'src/modules/solana/transactions/create';
 import { WithReadonlyWarningDialog } from './ReadonlyWarningDialog';
 
-type SendTxParams = TransactionContextParams & {
-  transaction: IncomingTransaction;
-};
+type SendTxParams = TransactionContextParams &
+  OneOf<{ transaction: IncomingTransaction; solTransaction: SolTransaction }>;
```

Comment:
can it be better to name transaction -> ethTransaction?

Reply:
Yes, ideally all these names and concepts should be refined

Currently we have drafted something like this in our new backend api:

```
type TransactionMultichain = {
  evm: null | TransactionEVM;
  solana: null | TransactionSolana;
};
```